recipes/android-ndk/all/conanfile.py

```diff
@@ -0,0 +1,207 @@
+        self.env_info.CHOST = self._llvm_triplet
+
+        ndk_sysroot = os.path.join(self._ndk_root, 'sysroot')
+        self.output.info('Creating CONAN_CMAKE_FIND_ROOT_PATH environment variable: %s' % ndk_sysroot)
+        self.env_info.CONAN_CMAKE_FIND_ROOT_PATH = ndk_sysroot
+
+        self.output.info('Creating SYSROOT environment variable: %s' % ndk_sysroot)
+        self.env_info.SYSROOT = ndk_sysroot
+
+        self.output.info('Creating self.cpp_info.sysroot: %s' % ndk_sysroot)
+        self.cpp_info.sysroot = ndk_sysroot
+
+        self.output.info('Creating ANDROID_NATIVE_API_LEVEL environment variable: %s' % self.settings_target.os.api_level)
+        self.env_info.ANDROID_NATIVE_API_LEVEL = str(self.settings_target.os.api_level)
+
+        self._chmod_plus_x(os.path.join(self.package_folder, "cmake-wrapper"))
+        cmake_wrapper = "cmake-wrapper.cmd" if self.settings.os == "Windows" else "cmake-wrapper"
+        cmake_wrapper = os.path.join(self.package_folder, cmake_wrapper)
+        self.output.info('Creating CONAN_CMAKE_PROGRAM environment variable: %s' % cmake_wrapper)
+        self.env_info.CONAN_CMAKE_PROGRAM = cmake_wrapper
+
+        toolchain = os.path.join(self.package_folder, "build", "cmake", "android.toolchain.cmake")
+        self.output.info('Creating CONAN_CMAKE_TOOLCHAIN_FILE environment variable: %s' % toolchain)
+        self.env_info.CONAN_CMAKE_TOOLCHAIN_FILE = toolchain
+
+        self.env_info.CC = self._define_tool_var('CC', 'clang')
+        self.env_info.CXX = self._define_tool_var('CXX', 'clang++')
+        self.env_info.LD = self._define_tool_var('LD', 'ld')
+        self.env_info.AR = self._define_tool_var('AR', 'ar')
+        self.env_info.AS = self._define_tool_var('AS', 'as')
+        self.env_info.RANLIB = self._define_tool_var('RANLIB', 'ranlib')
+        self.env_info.STRIP = self._define_tool_var('STRIP', 'strip')
+        self.env_info.ADDR2LINE = self._define_tool_var('ADDR2LINE', 'addr2line')
+        self.env_info.NM = self._define_tool_var('NM', 'nm')
+        self.env_info.OBJCOPY = self._define_tool_var('OBJCOPY', 'objcopy')
+        self.env_info.OBJDUMP = self._define_tool_var('OBJDUMP', 'objdump')
+        self.env_info.READELF = self._define_tool_var('READELF', 'readelf')
+        self.env_info.ELFEDIT = self._define_tool_var('ELFEDIT', 'elfedit')
+
+        self.env_info.ANDROID_PLATFORM = "android-%s" % self.settings_target.os.api_level
+        self.env_info.ANDROID_TOOLCHAIN = "clang"
+        self.env_info.ANDROID_ABI = self._android_abi
+        libcxx_str = str(self.settings_target.compiler.libcxx)
+        self.env_info.ANDROID_STL = libcxx_str if libcxx_str.startswith('c++_') else 'c++_shared'
+
+        self.env_info.CMAKE_FIND_ROOT_PATH_MODE_PROGRAM = "BOTH"
+        self.env_info.CMAKE_FIND_ROOT_PATH_MODE_LIBRARY = "BOTH"
+        self.env_info.CMAKE_FIND_ROOT_PATH_MODE_INCLUDE = "BOTH"
+        self.env_info.CMAKE_FIND_ROOT_PATH_MODE_PACKAGE = "BOTH"
```

Comment:
When cross compiling, these variables are normally set to `ONLY`.
I assume these are set to `BOTH` to allow finding files in external conan directories?

In practice, did you encounter problems with cmake finding packages for the wrong architecture? (e.g. for the build machine architecture instead for the android architecture)

Reply:
I don't know, I would have to try, I re used what worked 
What is with other package dependencies you have, that are compiled with this package, and required.
They are in the conan dir, so not in the root dir

Reply:
checked, it has to be BOTH, because we are looking in folders outside of the sysroot, like those that are provided for dependencies via conan.